include/deal.II/non_matching/mapping_info.h

```diff
@@ -241,7 +241,8 @@ namespace NonMatching
      */
     void
     reinit(const typename Triangulation<dim, spacedim>::cell_iterator &cell,
-           const Quadrature<dim> &quadrature);
+           const Quadrature<dim> &quadrature,
+           const bool             quadrature_weights_are_JxW = false);
```

Comment:
I am asking myself if this option is something that we know at the time when we construct the `MappingInfo` object. Maybe we need an `AdditionalData` object for `MappingInfo` in the future to enable behavior/functionality. What do you (and others) think? @jh66637 @peterrum @kronbichler

Reply:
I agree with @bergbauer that it would be better set this option in the constructor e.g. via an AD object.

Reply:
Also, I would rather say that global weights should be used (`use_global_weight`), as provided by `QSimplex::compute_affine_transformation()` -> comment.